notebook/static-src/base/js/dialog.js

```diff
@@ -162,47 +157,47 @@ define(function(require) {
                         .append(textarea)
                     )
             );
-        var editor = CodeMirror.fromTextArea(textarea[0], {
-            lineNumbers: true,
-            matchBrackets: true,
-            indentUnit: 2,
-            autoIndent: true,
-            mode: 'application/json',
-        });
-        var modal_obj = modal({
-            title: "Edit " + options.name + " Metadata",
-            body: dialogform,
-            buttons: {
-                OK: { class : "btn-primary",
-                    click: function() {
-                        /**
-                         * validate json and set it
-                         */
-                        var new_md;
-                        try {
-                            new_md = JSON.parse(editor.getValue());
-                        } catch(e) {
-                            console.log(e);
-                            error_div.text('WARNING: Could not save invalid JSON.');
-                            return false;
+        requirejs(['codemirror/lib/codemirror'], function(CodeMirror) {
```

Comment:
Are you able to explain or comment on why requirejs is used within the node source here?